system-tests/ft-suite/src/test/kotlin/com/hotels/styx/logging/HttpMessageLoggingSpec.kt

```diff
@@ -0,0 +1,118 @@
+import com.hotels.styx.startup.StyxServerComponents
+import com.hotels.styx.support.matchers.LoggingTestSupport
+import io.kotlintest.Spec
+import io.kotlintest.specs.FeatureSpec
+import reactor.core.publisher.toMono
+
+class HttpMessageLoggingSpec : FeatureSpec() {
+
+    init {
+        feature("Styx request/response logging") {
+
+            scenario("Logger should hide cookies and headers") {
+                val proxyHost = "${styxServer.proxyHttpAddress().hostName}:${styxServer.proxyHttpAddress().port}"
+
+                client.send(HttpRequest.get("/a/path")
+                        .header(HttpHeaderNames.HOST, proxyHost)
+                        .header("header1", "h1")
+                        .header("header2", "h2")
+                        .header("cookie", "cookie1=c1;cookie2=c2")
+                        .build())
+                        .toMono()
+                        .block()!!
+
+                val expectedRequest = Regex("requestId=[-a-z0-9]+, secure=false, origin=null, "
+                    + "request=\\{version=HTTP/1.1, method=GET, uri=/a/path, headers=\\[Host=localhost:[0-9]+, header1=\\*\\*\\*\\*, header2=h2, cookie=cookie1=\\*\\*\\*\\*;cookie2=c2\\], id=[-a-z0-9]+\\}")
+
+                val expectedResponse = Regex("requestId=[-a-z0-9]+, secure=false, "
+                    + "response=\\{version=HTTP/1.1, status=200 OK, headers=\\[header1=\\*\\*\\*\\*, header2=h2, cookie=cookie1=\\*\\*\\*\\*;cookie2=c2, Via=1.1 styx\\]\\}")
+
+                logger.log().shouldContain(INFO, expectedRequest)
+                logger.log().shouldContain(INFO, expectedResponse)
+            }
+        }
+    }
+
+    val logger = LoggingTestSupport("com.hotels.styx.http-messages.inbound")
+
+    val client: StyxHttpClient = StyxHttpClient.Builder().build()
+
+    val yamlText = """
+        proxy:
+          connectors:
+            http:
+              port: 0
+
+            https:
+              port: 0
+              sslProvider: JDK
+              sessionTimeoutMillis: 300000
+              sessionCacheSize: 20000
+              
+        request-logging:
+          inbound:
+            enabled: true
+            longFormat: true
+          hideCookies:
+            - cookie1
+          hideHeaders:
+            - header1
+
+        admin:
+          connectors:
+            http:
+              port: 0
+
+        routingObjects:
+          root:
+            type: StaticResponseHandler
+            config:
+              status: 200
+              content: ""
+              headers:
+                - name: "header1"
+                  value: "h1"
+                - name: "header2"
+                  value: "h2" 
+                - name: "cookie"
+                  value: "cookie1=c1;cookie2=c2"
+
+        httpPipeline: root
+      """.trimIndent()
+
+    val styxServer = StyxServer(StyxServerComponents.Builder()
+            .styxConfig(StyxConfig.fromYaml(yamlText))
+            .build())
```

Comment:
There is a class called `StyxServerProvider` in support/StyxServerProvider.kt. It provides a unique reference point to a Styx server while allowing it to be restarted, and thus cleared from any acquired state`, between the tests.

You don't really have to use it in this test, but could make this more consistent with others.